extensions/jms-spi/deployment/src/main/java/io/quarkus/jms/spi/deployment/ConnectionFactoryWrapperBuildItem.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.jms.spi.deployment;
+
+import java.util.function.Function;
+
+import javax.jms.ConnectionFactory;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+
+/**
+ * A build item that can be used to wrap the JMS ConnectionFactory
+ */
+public final class ConnectionFactoryWrapperBuildItem extends SimpleBuildItem {
```

Comment:
BTW, you aware that you will be able to produce only one of these?

Not sure exactly for what it is used but that probably means you won't be able to mix several JMS extensions (not sure if it makes sense).

Reply:
As far as I know, only [quarkus-pooled-jms](https://github.com/quarkiverse/quarkus-pooled-jms) will produce this BuildItem to provide pooling and JTA integration capabilities currently . Other JMS extensions can consume it.